### DISCUSSION TITLE
Execution error trying to require flow-storm

### DISCUSSION THREAD

**Discussion by kazlibrowski** [jpmonettas/flow-storm-debugger#48] [Q&A] [answered]
Saw a video. Looks fantastic. However, I have problems firing it up.

I run the command from bash as instructed:

clj -Sforce -Sdeps '{:deps {com.github.jpmonettas/flow-storm-dbg {:mvn/version "3.0.216"} com.github.jpmonettas/flow-storm-inst {:mvn/version **"3.0.216"}}}'

I get the repl, and from there I try to do the require:

user=> (require '[flow-storm.api :as fs-api])

I get this:
Execution error (NoSuchMethodError) at com.google.javascript.jscomp.RhinoErrorReporter/<clinit> (RhinoErrorReporter.java:169).
'com.google.common.collect.ImmutableMap com.google.common.collect.ImmutableMap$Builder.buildOrThrow()'

Any ideas?

**Comment by jpmonettas** [comment, marked as answer]
Hey thanks @kazlibrowski !

Looks like there is something else on the classpath that is conflicting with the debugger, maybe your ~/.clojure/deps.edn is bringing something in?
Can you paste the output of running `clj -Sforce -Sdeps '{:deps {com.github.jpmonettas/flow-storm-dbg {:mvn/version "3.0.216"} com.github.jpmonettas/flow-storm-inst {:mvn/version "3.0.216"}}}' -Stree` in the same folder it is failing?

I have seen that conflict before and it wasn't related to flow-storm, was related to clojurescript (which flow-storm depends on) and an old version of google guava.

**Reply by kazlibrowski** [reply]
Thanks for your quick reply. I had accidentally run the clj command in a directory that had a deps.edn file in it. Once I moved to a different, empty directory, I managed to get a lot further. I'm still not there though. I'm on WSL2 and the error I now have is "Can't connect to X11 window server using '172.31.0.1:0' as the value of the DISPLAY variable.". I'll figure out how to set up the X-Server appropriately in the Windows environment and post up some notes to help others.

**Reply by kazlibrowski** [reply]
For those like me using this through WSL2 on Windows. You'll need to set up an X-Server on Windows, make sure you drill a hole in the firewall for the port, and then specify the display for the WSL process. Then everything should work. The steps are:

1) Install VcXsrv on Windows.
2) Run the XLaunch app on Windows, Choose Multiple Windows, Display Number 0, Start no client, Check all settings on the Extra Settings screen (specifically "Disable access control"
3) In the WSL2 terminal, run ip addr | grep eth0 to determine the ip of the xserver
4) On Windows, go to Firewall and network protection, select Advanced Settings, and add a new Inbound Rule in the Windows Defender window that pops up.
5) For the rule, select Port, then TCP, specific port 6000, then click next twice, and finally name the rule something appropriate, like "XServer rule".
6) Now find the rule you just created, right click, select Properties, then the Scope tab, and enter the IP address you found at step 3 with an appropriate range in order to allow the port through for the WSL2 subsystem.
7) Now find the ip address of your windows machine by typing ipconfig in a windows terminal
8) Now, in the WSL2 terminal, type export DISPLAY=<windows ip adddress>:0.0
9) Also in the WSL2 terminal, type export LIBGL_ALWAYS_INDIRECT=1
10) Now you can start clojure with the command clj -Sforce -Sdeps '{:deps {com.github.jpmonettas/flow-storm-dbg {:mvn/version "3.0.216"} com.github.jpmonettas/flow-storm-inst {:mvn/version "3.0.216"}}}'
11) That should get you into the REPL, where you should type (require '[flow-storm.api :as fs-api])
12) Finally, type (fs-api/local-connect) and you're done!

Yes, it's a bit of a faff, but absolutely worth it to get this debugger running on WSL2. It's a great demonstration of something that Clojure can do but which would be REALLY hard to achieve in other languages.

**Reply by jpmonettas** [reply]
Oh nice, thanks a lot for those detailed steps, I'll add them to the docs!

**Reply by kazlibrowski** [reply]
Just a quick follow-up to tie this off completely.
First of all, something went wrong with my pasting. In step 8, you need to put in the ip address you found in step 7, so something like:
export DISPLAY=192.168.1.43:0.0

Also, you were right, my initial problem was a dependency conflict with google guava, which was indirectly referenced by com.auth0/java-jwt. Had you not given me the guava tip, I'm not sure I'd have found this very easily in my 497 line dependency tree output. We need another class of debugger to sort out dependency hell :-) Fortunately, I can make do without the auth0 library so I'm finally up and running with a "serious" project to debug.

**Reply by jpmonettas** [reply]
Yeah the guava thing was tricky, and those are hard to debug. If you still want to use your java-jwt you just need to force guava to the latest version, the conflict seems to go away and it hopefully doesn't break java-jwt. Mabe even the latest version of java-jwt already uses the latest guava, no idea.

Thank again for all the instructions for WSL2!

**Reply by jpmonettas** [reply]
Just added your instructions here https://jpmonettas.github.io/flow-storm-debugger/user_guide.html#_working_on_windows_with_wsl2
So if you need to repeat them they are on the users guide now.
